### PR TITLE
more security hardening

### PR DESCRIPTION
**Pull request by ComputerGenius152**
- add csrf tokens to prevent http://127.0.0.1:5000/changepassword?password=password123&newPassword=123password&confirmation=123password
- let users decide on samesite cookies
- ahh, security
image

### CODE REVIEW

**Comment by ComputerGenius152** [comment]
- make jquery files local, with cache partitioning there is no longer benefit to a cdn

File: application.py [hunk 1]
```diff
@@ -36,11 +37,16 @@
...
+csrf = CSRFProtect(app)
+csrf.init_app(app)
+
 @app.before_request
 def check_for_maintenance():
     # crappy if/elses used here for future expandability
     global maintenance_mode
-    if session:
+    # don't block the user if they only have the csrf token
+    if ("csrf_token" in session and len(session) > 1) or (not "csrf_token" in session and len(session)):
```
Comment (jdabtieu) [inline application.py:49]: this will throw an exception if session doesn't exist

Comment (jdabtieu) [reply application.py:49]: Otherwise, looks good

**Review by jdabtieu — CHANGES_REQUESTED** [review]
See comments

File: application.py (hunk 1 above)
Comment (jdabtieu) [reply application.py:49]: Nevermind, it works somehow

**Review by jdabtieu — APPROVED** [review]

**Pull request merged**